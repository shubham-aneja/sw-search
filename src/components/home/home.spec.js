import React from 'react';
import { mount } from 'enzyme';

import Home from './index.js';
import Planet from '../planet'
import NoResult from '../no-result'

describe('Home', () => {
    it('rendering', () => {
        const onValueChangeMock = jest.fn()
        const doLogoutMock = jest.fn()
        const homeDestroyMock = jest.fn()
        const props = {
            onValueChange: onValueChangeMock,
            doLogout: doLogoutMock,
            homeDestroy: homeDestroyMock,
            isLoading: false,
            options: [],
            searchedText: '',
            isNoResult: false,
            error: ''
        };
        const app = mount(<Home {...props} />)
        expect(app.html()).toMatchSnapshot();

        const inputs = app.find('input')
        const userNameInput = inputs.at(0)
        userNameInput.simulate('change', {target: {value: 'Pla'}})

        app.setProps({
            options: [
                {name: 'Planet 1', population: 10000},
                {name: 'Planet 2', population: 1000000},
                {name: 'Planet 3', population: 'unknown'},
            ]
        });

        app.setProps({
            error: 'Something went wrong.'
        })
    });

    it('Option renderer', () => {
        expect(Planet({
            name: 'Keplin',
            population: 1000
        })).toMatchSnapshot()
        expect(Planet({
            name: 'Pluto',
            population: 'unknown'
        })).toMatchSnapshot()
    })
});

