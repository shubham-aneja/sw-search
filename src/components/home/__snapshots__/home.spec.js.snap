// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Option renderer 1`] = `
<div
  className="planet__container population-level-1"
>
  <div
    className="planet__name"
  >
    Keplin
     
  </div>
  <div
    className="planet__population"
  >
    Having population:
    <span
      className="planet__population-count"
    >
      1000
    </span>
  </div>
</div>
`;

exports[`Home Option renderer 2`] = `
<div
  className="planet__container population-level-undefined"
>
  <div
    className="planet__name"
  >
    Pluto
     
  </div>
  <div
    className="planet__population"
  >
    Having population:
    <span
      className="planet__population-count"
    >
      unknown
    </span>
  </div>
</div>
`;

exports[`Home rendering 1`] = `"<div class=\\"home-app\\"><div class=\\"header-bar\\"><h2>Home</h2><div class=\\"action-button-logout\\"><span>Logout</span></div></div><input type=\\"text\\" placeholder=\\"start search by typing...\\" value=\\"\\"><div class=\\"options-container\\"></div><div class=\\"error\\"></div></div>"`;
